apps/call.go

```diff
@@ -131,6 +131,9 @@ type CallResponse struct {
 
 	// Used in CallResponseTypeForm
 	Form *Form `json:"form,omitempty"`
+
+	// Used to provide info about the App to client, e.g. the bot user id
+	AppMetadata AppMetadataForClient `json:"app_metadata"`
```

Comment:
1/5 I'd rather avoid adding private fields (i.e. irrelevant to the app) to the CallResponse. Can we wrap `CallResponse` into a `ProxyCallResponse` perhaps?

0/5 I am also contemplating that `apps.Request` and `apps.Response` maybe better names.

Reply:
> 1/5 I'd rather avoid adding private fields (i.e. irrelevant to the app) to the CallResponse. Can we wrap `CallResponse` into a `ProxyCallResponse` perhaps?

So we would use `CallResponse` everywhere that we are handling what the app returned, but convert to a `ProxyCallResponse` when we are packaging it up for the UI?

> 0/5 I am also contemplating that `apps.Request` and `apps.Response` maybe better names.

Would this change apply to the redux types as well? @larkox What are your thoughts on this?

Reply:
redux types are used (should be used?) by the apps, so we'd need to have both in redux, yes; but `Proxy...` is not for App developers.

Reply:
Long term ideal solution:
- Have two types file for Apps: package exported types (for apps developers) and internally used types (not exported outside of the package).
- ProxyResponse be a struct with the Response Embedded
- ProxyResponse to have a custom Marshaller to flatten the struct (I don't like the idea of having to handle in the client something like `response = {appData: {...}, response: {...}}`)

How does this sound?

Reply:
@larkox Wrapping the `CallResponse` inside the `ProxyCallResponse` doesn't require a special Marshaller. We can use an embedded struct like so:

```go
type ProxyCallResponse struct {
	*CallResponse

	AppMetadata *AppMetadataForClient `json:"app_metadata"`
}
```

Reply:
For some reason I thought the marshaller would not flatten the list. LGTM.